index.js

```diff
@@ -355,9 +351,7 @@ class AddApiKey {
   constructor(serverless, options) {
     this.options = options;
     this.hooks = {
-      'after:deploy:deploy': async function () {
-        await addApiKey(serverless, options);
-      }
+      'after:deploy:deploy': () => addApiKey(serverless, options)
```

Comment:
@abeMedia this function needs to be `async`.

Reply:
Hi @rrahul963, an `async` function always returns a promise but since your function already returns a promise it's exactly the same when removing it.

Reply:
in past i have seen that when the function was not `async`.. the code was not getting executed properly..

Reply:
Was the function you were calling returning a `Promise` though? Because `async/await` is just syntactic sugar for functions that returns a `Promise`, avoiding the `.then()` chains.

See the function signature of your `async` function (returns `Promise<void>`):
![image](https://user-images.githubusercontent.com/4355715/54363630-2bad6000-4663-11e9-8079-bbf5e5396637.png)
Here you can see since I'm returning the function directly the wrapping function still returns `Promise<void>`.
![image](https://user-images.githubusercontent.com/4355715/54363873-bd1cd200-4663-11e9-94ca-c60d5ba765f8.png)
And, just to demostrate it would be the same if the wrapper was an async:
![image](https://user-images.githubusercontent.com/4355715/54363986-f6554200-4663-11e9-92c4-350d71c3d641.png)